odoo/addons/base/models/res_partner.py

```diff
@@ -112,9 +112,10 @@ class PartnerCategory(models.Model):
     def _get_default_color(self):
         return randint(1, 11)
 
-    name = fields.Char(string='Tag Name', required=True, translate=True)
-    color = fields.Integer(string='Color', default=_get_default_color)
-    parent_id = fields.Many2one('res.partner.category', string='Parent Category', index=True, ondelete='cascade')
+    name = fields.Char('Name', required=True, translate=True)
+    display_name = fields.Char('Display Name', compute='_compute_display_name', store=True)
```

Comment:
You should not store display_name, as it depends on language: you will store a version that depends on current user's lang and it is not consistent. Did you check filtering on child_ids instead ?

Reply:
@tde-banana-odoo yes i did and it doesn't search anything. i used this filter domain: `filter_domain="[('child_ids', 'ilike', self)]"`. and i tried it on newly fresh database. 
![image](https://github.com/odoo/odoo/assets/52849166/42558fdb-08f3-47ea-a589-48d4c4dbd384)
![image](https://github.com/odoo/odoo/assets/52849166/68a9eb26-4268-4af0-a440-c7a56d08e08d)

Reply:
i don't know how it's implemented, but results came out strange

Reply:
I don't think you can possible write a single query for that. It has to happen in two stages:
- find the tags with the matching name
- get its children

All that because tags can have multiple levels of parents, I don't think there's any domain operator you can write to do the equivalent of:
`['|', ..., ('name', 'ilike', value), ('parent_id.name', 'ilike', value), ('parent_id.parent_id.name', 'ilike', value), ...]`

As I remember the custom search method worked well without storing it. But indeed you can't apply anything but "ilike" to it.

Seems like you'll need some different approach if we want an accurate search

Reply:
@tde-banana-odoo what should i do here?

Reply:
Side remark: ``_name_search`` is the method used when searching based on name and should be the complement to ``display_name`` implementation. Which is why name_search explodes "/" to retake the last part, as display_name concatenates the path using "/" .

Here we are not really doing a name_search: more a search on children or ancestors. Actually looking at the code and the spec, I am unsure what we want: ancestors, children, both ?

Reply:
@tde-banana-odoo we want to search for a children:
```
A
└─ B
   └─ C
 ```
 when i seach for category A, it should return A, B, and C

Reply:
if i search for B category, it should return B and C

Reply:
And using inselect ? Something copy pasted from other code, to test (did not really play with them since they have been added to odoo so clearly not an expert)
```
    query = SQL("""
        SELECT id
        FROM {self._table}
        WHERE {condition}
    """, params)
    return [('id', 'inselect, query)]
```

Reply:
Or
```
return [
    ('id',
     'child_of',
     self.env['res.partner.category']._search([('name', 'ilike', value)])
)]
```

Reply:
But all that has to be in a search method, indeed. I wanted to use a filter_domain but don't think it is possible to do that with a "simple" domain. Sorry for spamming.

Reply:
@tde-banana-odoo i did the second solution, but not sure what's the difference betrween mine and that one (only one difference is that i am using private search, nothing else)